cogs/roles.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+from os import getenv
+from typing import TYPE_CHECKING
+
+from nextcord import Interaction, Member, Object, SelectOption, slash_command
+from nextcord.ext.commands import Cog
+from nextcord.ui import Select, View
+from nextcord.utils import get
+
+if TYPE_CHECKING:
+    from nextcord.abc import Snowflake
+
+GUILD_ID = int(getenv("GUILD_ID", 0))
+ASSIGNABLE_ROLE_IDS = {int(r) for r in getenv("ASSIGNABLE_ROLE_IDS", "0,0").split(",")}
+
+
+class RolesView(View):
+    def __init__(self, *, member: Member | None):
+        super().__init__(timeout=None)
+
+        self.add_item(RolesSelect(member=member))
+
+
+class RolesSelect(Select["RolesView"]):
+    def __init__(self, *, member: Member | None):
+        # this is being invoked to add persistency
+        # we only care about custom_id for the store
+        # we cannot use guild.me as the bot is not ready so a guild is not available
+        if member is None:
+            return super().__init__(
+                custom_id="roles:select", options=[SelectOption(label="placeholder")]
+            )
```

Comment:
Dont allow no member as it will cause issues

Reply:
what kind of issues, what do you propose as a persistent view fix that will work with this current setup